bindings/python/iota_sdk/types/address.py

```diff
@@ -2,42 +2,26 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from iota_sdk.types.common import HexStr
-from enum import Enum
+from dataclasses import dataclass
+from enum import IntEnum
+from typing import Optional
 
 
-class AddressType(Enum):
+class AddressType(IntEnum):
     ED25519 = 0
     ALIAS = 8
     NFT = 16
 
 
+@dataclass
 class Address():
-    def __init__(self, type: AddressType, address_or_id: HexStr):
-        """Initialize an Address
-
-        Parameters
-        ----------
-        type : AddressType
-            The type of the Address
-        address_or_id : string
-            The address to use. Can either be an hex encoded ED25519 address or NFT/Alias id
-        """
-        self.type = type
-        self.address_or_id = address_or_id
+    type: int
+    pubKeyHash: Optional[HexStr] = None
+    aliasId: Optional[HexStr] = None
+    nftId: Optional[HexStr] = None
```

Comment:
having this here instead of in their respective classes is just to make the as_dict easier right?

Reply:
same for feature, i guess.

Reply:
Also for the deserialization
Doesn't really look good, but it works and I don't know how else to do it better